### PR TITLE
refactor: Convert Button stories to csf format

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Converted stories for `ButtonLink`, `ButtonPrimary`, `ButtonSecondary`, and `ButtonIcon`
- Removed story for `ButtonBase`
- Added sample const to `ButtonIcon`
- Updated consts for `ButtonIcon`

## **Screenshots/Recordings**
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 30 04
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 30 06
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 30 16
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 30 18
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 30 25
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 30 27
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 30 30
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 30 32
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 37 33
Simulator Screen Shot - iPhone 14 Pro - 2023-10-04 at 18 37 35

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Buttons/Button/variants/ButtonPrimary/ButtonPrimary.stories.tsx [hunk 1]
```diff
@@ -1,20 +1,53 @@
...
+      control: { type: 'boolean' },
+      defaultValue: SAMPLE_BUTTONPRIMARY_PROPS.isDanger,
+    },
+    width: {
+      options: ButtonWidthTypes,
+      control: {
+        type: 'select',
+      },
+      defaultValue: SAMPLE_BUTTONPRIMARY_PROPS.width,
+    },
```
Comment (georgewrmarshall) [inline app/component-library/components/Buttons/Button/variants/ButtonPrimary/ButtonPrimary.stories.tsx:48]: Missing prop controls: `labelColor`, `isDisabled`

File: app/component-library/components/Buttons/Button/foundation/ButtonBase/ButtonBase.stories.tsx
Comment (georgewrmarshall) [inline app/component-library/components/Buttons/Button/foundation/ButtonBase/ButtonBase.stories.tsx:1]: I still feel there is value in keeping the ButtonBase story for future iterations in isolation?

File: app/component-library/components/Buttons/Button/Button.stories.tsx [hunk 2]
```diff
@@ -1,65 +1,57 @@
...
+  title: 'Component Library / Buttons',
+  component: ButtonComponent,
+  argTypes: {
+    variant: {
+      options: ButtonVariants,
+      control: {
+        type: 'select',
+      },
+      defaultValue: ButtonVariants.Primary,
+    },
```
Comment (georgewrmarshall) [inline app/component-library/components/Buttons/Button/Button.stories.tsx:24]: Do we want to add all the other button prop controls here?

https://github.com/MetaMask/metamask-mobile/assets/8112138/431afe64-daf7-4498-89f4-ccf0da710cad

File: app/component-library/components/Buttons/ButtonIcon/ButtonIcon.stories.tsx [hunk 3]
```diff
@@ -1,57 +1,42 @@
...
+      },
+      defaultValue: SAMPLE_BUTTONICON_PROPS.iconName,
+    },
+    variant: {
+      options: ButtonIconVariants,
+      control: {
+        type: 'select',
+      },
+      defaultValue: SAMPLE_BUTTONICON_PROPS.variant,
+    },
```
Comment (georgewrmarshall) [inline app/component-library/components/Buttons/ButtonIcon/ButtonIcon.stories.tsx:26]: I think this is just a disparity between platforms but ButtonIcon doesn't have variants in Figma or Extension

**Review by georgewrmarshall — COMMENTED** [review]
Looking good. Left some comments

**Review by georgewrmarshall — DISMISSED** [review]

File: app/component-library/components/Buttons/Button/variants/ButtonPrimary/ButtonPrimary.stories.tsx (hunk 1 above)
Comment (brianacnguyen) [reply app/component-library/components/Buttons/Button/variants/ButtonPrimary/ButtonPrimary.stories.tsx:48]: `labelColor` will be omitted

File: app/component-library/components/Buttons/Button/foundation/ButtonBase/ButtonBase.stories.tsx
Comment (brianacnguyen) [reply app/component-library/components/Buttons/Button/foundation/ButtonBase/ButtonBase.stories.tsx:1]: i dont want to encourage the use of button base so will leave them out

File: app/component-library/components/Buttons/Button/Button.stories.tsx (hunk 2 above)
Comment (brianacnguyen) [reply app/component-library/components/Buttons/Button/Button.stories.tsx:24]: will try to re-add that in at a later date

File: app/component-library/components/Buttons/ButtonIcon/ButtonIcon.stories.tsx (hunk 3 above)
Comment (brianacnguyen) [reply app/component-library/components/Buttons/ButtonIcon/ButtonIcon.stories.tsx:26]: we'll need to have a component sync on this

**Review by georgewrmarshall — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**